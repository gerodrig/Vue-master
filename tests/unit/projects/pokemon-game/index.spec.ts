import {beforeEach, describe, expect, it, vi } from 'vitest';
import { shallowMount, mount } from '@vue/test-utils';

import PokemonComponent from '../../../../src/projects/pokemon-game/index.vue'
import { pokemons } from '../../mocks/pokemon-game/pokemons.mock';


describe('Pokemon index component', () => {

  let wrapper: any;
  let localStorageMock;

  beforeEach(() => {
    //local storage mock
    localStorageMock = {
      getItem: vi.fn(),
      setItem: vi.fn(),
      clear: vi.fn(),
    };
    global.localStorage = localStorageMock;
    wrapper = shallowMount(PokemonComponent);
  });

  it('must match the snapshot', () => {
    
    expect(wrapper.html()).toMatchSnapshot();
    
  });

  it('must get options when mounting component', async () => {

    const wrapper = shallowMount(PokemonComponent, {
      setup() {
        return {
          options: pokemons,
          answer: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: '',
          won: false,
          checkAnswer: () => {},
          resetGame: () => {},
        };
      },
    });

    const options = wrapper.vm.options;

    // console.log(wrapper.vm.options);
    expect(options.length).toBe(4);

    
  });

  it('should match the snapshot when pokemons are loaded', () => {
      const wrapper = mount(PokemonComponent, {
        data() {
          return {
            options: pokemons,
            answer: pokemons[0],
            showPokemon: false,
            showAnswer: false,
            message: '',
            won: false,
            checkAnswer: () => {},
            resetGame: () => {},
          };
        },
      });

      expect(wrapper.html()).toMatchSnapshot();
  });

  it('must show PokemonPicture and PokemonOptions must exist', () => {
    const wrapper = shallowMount(PokemonComponent, {
      setup() {
        return {
          options: pokemons,
          answer: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: '',
          won: false,
          checkAnswer: () => {},
          resetGame: () => {},
        };
      },
    });
    const pokemonOptions = wrapper.find('pokemon-options-stub');
    const picture = wrapper.find('pokemon-picture-stub');

    expect(picture.exists()).toBeTruthy();
    expect(picture.exists()).toBeTruthy();
    //pokemon picture must have attribute 1
    expect(picture.attributes('pokemonid')).toBe('1');

    // check that options in PokemonOptions exists
    expect(pokemonOptions.exists()).toBeTruthy();
  });

  it('tests the checkAnswer method', async () => {

    const wrapper = mount(PokemonComponent, {
      setup() {
        return {
          options: pokemons,
          answer: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: '',
          won: false,
          checkAnswer: (num: number) => 1,
          resetGame: () => {},
        };
      },
    });

    await wrapper.vm.checkAnswer(1);
    // check that the message is empty
    expect(wrapper.vm.message).toBe('');   

    // check that the showAnswer is false
    expect(wrapper.vm.showAnswer).toBeFalsy();

    //check pokemon answer
    expect(wrapper.vm.answer).toEqual({ id: 1, name: 'Bulbasaur' });
    
    
  });
});