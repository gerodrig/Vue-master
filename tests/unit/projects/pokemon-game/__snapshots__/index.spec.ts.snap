// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pokemon index component > must match the snapshot 1`] = `
"<div class=\\"relative w-full h-[80vh] md:h-screen bg-cover bg-[url(@/assets/pokemon-game/background.png)] transform -translate-y-36\\">
  <pokemon-picture-stub pokemonid=\\"0\\" showpokemon=\\"false\\" defaultpath=\\"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/\\" class=\\"absolute left-28 top-64 md:left-56 md:top-56 xl:left-1/4 xl:top-1/3\\"></pokemon-picture-stub>
  <game-notifications-stub message=\\"Who is that pokemon?\\" won=\\"false\\" showanswer=\\"false\\"></game-notifications-stub>
</div>
<div class=\\"-mt-36\\">
  <pokemon-options-stub options=\\"\\" disabled=\\"false\\"></pokemon-options-stub>
</div>"
`;

exports[`Pokemon index component > should match the snapshot when pokemons are loaded 1`] = `
"<div class=\\"relative w-full h-[80vh] md:h-screen bg-cover bg-[url(@/assets/pokemon-game/background.png)] transform -translate-y-36\\">
  <div data-v-de2a09f6=\\"\\" class=\\"absolute left-28 top-64 md:left-56 md:top-56 xl:left-1/4 xl:top-1/3\\"><img data-v-de2a09f6=\\"\\" src=\\"https://upload.wikimedia.org/wikipedia/commons/3/3b/MissingNo.svg\\" alt=\\"\`Pokemon ?\`\\" class=\\"hidden-pokemon\\" style=\\"display: none;\\"><!-- <button @click=\\"toggleReveal\\">Toggle Hide</button> -->
  </div>
  <div class=\\"absolute top-36 right-12 md:right-36 md:top-1/3 xl:right-24 xl:top-1/3 2xl:right-1/4 2xl:w-96 2xl:h-96\\">
    <!--v-if-->
  </div>
</div>
<div class=\\"-mt-36\\">
  <div data-v-6c7a75d2=\\"\\" class=\\"bg-black opacity-80\\">
    <div data-v-6c7a75d2=\\"\\" class=\\"flex items-center mx-auto justify-center max-w-[800px]\\">
      <ul data-v-6c7a75d2=\\"\\" class=\\"flex-wrap justify-center py-6 text-blue-700 list-none md:flex\\"></ul>
    </div>
  </div>
</div>"
`;
